### PR TITLE
refactor: migrate button

### PR DESCRIPTION
**Pull request by kyubisation**
This PR migrates and refactors the button (and link) module. This is a
major refactoring which completely changes the button and link modules
and components. See the new button documentation for details.

### CODE REVIEW

File: src/angular/styles/_typography.scss [hunk 1]
```diff
@@ -868,6 +868,320 @@ fieldset,
...
+  &:not(:disabled),
+  &:not(.sbb-disabled) {
+    &:hover,
+    &:focus {
+      background: $sbbColorRed125;
+    }
+  }
+
+  &:disabled,
+  &.sbb-disabled {
```
[the remark this answers is not in the document]
Comment (jeripeierSBB) [reply src/angular/styles/_typography.scss:978]: if yes, also other variants should be changed

**Review by jeripeierSBB — CHANGES_REQUESTED** [review]
First of all thanks for your great work! :-)

I think you missed to migrate showcase-merge to new buttons?

File: src/components-examples/angular/button/button-overview/button-overview-example.css [hunk 2]
```diff
@@ -0,0 +1,12 @@
+:host-context(html:not(.sbb-lean)) .sbb-lean-only {
```
[the remark this answers is not in the document]
Comment (kyubisation) [reply src/components-examples/angular/button/button-overview/button-overview-example.css:1]: Will change to warning and add notice for hidden disabled button

File: src/angular/styles/_typography.scss [hunk 3]
```diff
@@ -868,6 +868,320 @@ fieldset,
...
+  // in order to avoid tearing on state changes.
+  padding: pxToRem(9) pxToRem(39) pxToRem(11);
+  margin: 0;
+  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
+  outline: none;
+
+  @include mq($from: tabletPortrait) {
+    width: auto;
+    // We use em here, in order to scale for 4k and 5k
+    min-width: toEm(60 / $sizeFontDefault);
```
[the remark this answers is not in the document]
Comment (kyubisation) [reply src/angular/styles/_typography.scss:917]: check if pxToEm exists

File: src/angular/button/button.md [hunk 4]
```diff
@@ -5,6 +5,11 @@ Native `<button>` and `<a>` elements are always used in order to provide the mos
 and accessible experience for users. A `<button>` element should be used whenever some _action_
 is performed. An `<a>` element should be used whenever the user will _navigate_ to another view.
 
+```html
+<button type="button" sbb-button>Button</button>
+<a routerLink="..." sbb-button>Button</button>
```
Comment (jeripeierSBB) [inline src/angular/button/button.md:10]: I'd use sbb-link here, not?

**Review by jeripeierSBB — CHANGES_REQUESTED** [review]

**Review by jeripeierSBB — APPROVED** [review]
LGTM :)

**Pull request merged**